test: Fix failing net-next tests after changing to k8s 1.23

brb opened the pull request:
See commit msgs.

brb committed:
test: Disable CES for Egress GW suite

We need to disable CiliumEndpointSlice (CES) until [1] has been
resolved. Otherwise, once we switch the net-next CI job's k8s vsn to
1.23, the Egress GW suite will start failing.

Also, make sure that we don't unconditionally enable the CES if user
instructs to not enable.

[1]: https://github.com/cilium/cilium/issues/17669

Reported-by: Paul Chaignon <paul@cilium.io>
Signed-off-by: Martynas Pumputis <m@lambda.lt>

brb committed:
test: Fix --dry-run usage in ip-masq-agent tests

Since the test is going to be run on k8s 1.23, we need to pass the
"client" value to the "--dry-run" flag which was changed on k8s > 1.18.

Signed-off-by: Martynas Pumputis <m@lambda.lt>

brb committed:
test: Fix svc validation of ClusterIPs

Previously, the validateCiliumSvc() routine was assuming that ClusterIP
can be set only in a dedicated field, and it didn't consider the list of
ClusterIPs which are set when the DualStack is enabled.

The validateCiliumSvc() check gives very little value and it can be
removed. To avoid adding more variables to the net-next switch to k8s
1.23, let's do it in a separate PR.

Signed-off-by: Martynas Pumputis <m@lambda.lt>

pchaigno commented on test/helpers/kubectl.go:
```diff
@@ -2413,7 +2413,9 @@ func (kub *Kubectl) overwriteHelmOptions(options map[string]string) error {
 
-	if CiliumEndpointSliceFeatureEnabled() {
+	if _, found := options["enableCiliumEndpointSlice"]; !found &&
+		CiliumEndpointSliceFeatureEnabled() {
```
IIRC, the K8s>=1.21 was set arbitrarily, to not run everywhere in CI. So this helper can now be removed since it's disabled by default anyway. Do we even have any tests running with `enableCiliumEndpointSlice`?

pchaigno replied:
Previous discussion was at https://github.com/cilium/cilium/pull/17698#discussion_r736766323.

brb replied:
- The 4.9 job runs K8s 1.16 => CES disabled.
- The 4.19 runs K8s 1.22 => CES enabled for all.
- The net-next runs K8s 1.23 => CES enabled for all except for the Egress GW suite.

Do I miss something?

pchaigno replied:
Oh, no. I'm the one who missed something :facepalm: 
Thanks!

pchaigno reviewed (changes requested):
Sorry, missed something.

brb committed:
test: Skip graceful term tests if no kube-proxy

The graceful termination suite disables KPR (only bpf_lxc per-packet LB
is enabled for pod netns), which makes the K8s cluster half-functioning
if there is no kube-proxy installed. E.g., DNS resolution via CoreDNS
from host netns will fail.

The suite should be tested by 4.19 considering that it runs K8s 1.22.

Signed-off-by: Martynas Pumputis <m@lambda.lt>

brb committed:
test: Remove Cilium after pods deleted in K8sBandwidthTest

Previously, after running the K8sBandwidthTest suite, the netperf pods
were removed after Cilium agent removal which could lead to "Pods are
still not deleted after a timeout: 4m0s timeout expired".

The timeout was due to kubelet failing to find the Cilium's CNI binary:

    error killing pod: failed to "KillPodSandbox" for
    X with KillPodSandboxError: "rpc
    error: code = Unknown desc = networkPlugin cni failed to teardown pod
    "netperf-pod" network: failed to find plugin
    "cilium-cni" in path [/opt/cni/bin]"

Fix this by removing Cilium only after removing the pods.

Signed-off-by: Martynas Pumputis <m@lambda.lt>

pchaigno commented on test/k8sT/Bandwidth.go:
```diff
@@ -30,8 +30,12 @@ var _ = Describe("K8sBandwidthTest", func() {
 		ExpectAllPodsTerminated(kubectl)
+
+		UninstallCiliumFromManifest(kubectl, ciliumFilename)
+		ExpectAllPodsTerminated(kubectl)
```
Nice catch!

Why are we only seeing this now? Should we backport this fix?

brb replied:
I couldn't figure out which K8s change might have caused this :man_shrugging: 

Anyway, good idea re backporting. I will do it after it gets merged.

pchaigno reviewed (approved):
LGMT!
